docs/rules/object-curly-newline.md

```diff
@@ -0,0 +1,473 @@
+    bar: 2};
+let e = {
+    foo: function() {
+        dosomething();
+    }
+};
+
+let {
+} = obj;
+let {
+    f
+} = obj;
+let {g, h} = obj;
+let {i,
+    j} = obj;
+let {
+    k = function() {
+        dosomething();
+    }
+} = obj;
+```
+
+Examples of **correct** code for this rule with the `{"minProperties": 2}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"minProperties": 2}]*/
+/*eslint-env es6*/
+
+let a = {};
+let b = {foo: 1};
+let c = {
+    foo: 1, bar: 2
+};
+let d = {
+    foo: 1,
+    bar: 2
+};
+let e = {foo: function() {
+    dosomething();
+}};
+
+let {} = obj;
+let {f} = obj;
+let {
+    g, h
+} = obj;
+let {
+    i,
+    j
+} = obj;
+let {k = function() {
+    dosomething();
+}} = obj;
+```
+
+### a combination of multiline and minProperties
+
+Examples of **incorrect** code for this rule with the `{"multiline": true, "minProperties": 2}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"multiline": true, "minProperties": 2}]*/
+/*eslint-env es6*/
+
+let a = {
+};
+let b = {
+    foo: 1
+};
+let c = {foo: 1, bar: 2};
+let d = {foo: 1,
+    bar: 2};
+let e = {foo: function() {
+    dosomething();
+}};
+
+let {
+} = obj;
+let {
+    f
+} = obj;
+let {g, h} = obj;
+let {i,
+    j} = obj;
+let {k = function() {
+    dosomething();
+}} = obj;
+```
+
+Examples of **correct** code for this rule with the `{"multiline": true, "minProperties": 2}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"multiline": true, "minProperties": 2}]*/
+/*eslint-env es6*/
+
+let a = {};
+let b = {foo: 1};
+let c = {
+    foo: 1, bar: 2
+};
+let d = {
+    foo: 1,
+    bar: 2
+};
+let e = {
+    foo: function() {
+        dosomething();
+    }
+};
+
+let {} = obj;
+let {f} = obj;
+let {
+    g, h
+} = obj;
+let {
+    i,
+    j
+} = obj;
+let {
+    k = function() {
+        dosomething();
+    }
+} = obj;
+```
+
+### separating configuration
+
+Examples of **incorrect** code for this rule with the `{"literal": "always", "pattern": "never"}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"literal": "always", "pattern": "never"}]*/
+/*eslint-env es6*/
+
+let a = {};
+let b = {foo: 1};
+let c = {foo: 1, bar: 2};
+let d = {foo: 1,
+    bar: 2};
+let e = {foo: function() {
+    dosomething();
+}};
+
+let {
+} = obj;
+let {
+    f
+} = obj;
+let {
+    g, h
+} = obj;
+let {
+    i,
+    j
+} = obj;
+let {
+    k = function() {
+        dosomething();
+    }
+} = obj;
+```
+
+Examples of **correct** code for this rule with the `{"literal": "always", "pattern": "never"}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"literal": "always", "pattern": "never"}]*/
+/*eslint-env es6*/
+
+let a = [
+];
+let b = [
+    1
+];
+let c = [
+    1, 2
+];
+let d = [
+    1,
+    2
+];
+let e = [
+    function() {
+        dosomething();
+    }
+];
+
+let [] = obj;
+let [f] = obj;
+let [g, h] = obj;
+let [i,
+    j] = obj;
+let [k = function() {
+    dosomething();
+}] = obj;
+```
+
+## When Not To Use It
```

Comment:
Can you add a section for compatibility?

## Compatibility 

* **JSCS: rule name